### ISSUE TITLE
Incremental compilation doesn't work after moving sources with `semanticdbEnabled`

### ISSUE THREAD

**Issue by kubukoz** [sbt/sbt#6670]
- sbt version: 1.5.2 / 1.5.5 / 1.6.0-M1 (tested all)

- JVMs: `OpenJDK Runtime Environment GraalVM CE 21.2.0 (build 11.0.12+6-jvmci-21.2-b08)` / `OpenJDK Runtime Environment (Zulu 8.54.0.21-CA-macosx) (build 1.8.0_292-b10)` (tested all)
- OS: macOS Catalina 10.15.7

## steps

1. Create a project with `semanticDbEnabled := true`

```scala
lazy val root = (project in file("."))
  .settings(
    scalaVersion := "2.13.6",
    semanticdbEnabled := true,
  )
```

2. Add a Scala file to the sources (it can be anything). I made a full repository: https://github.com/kubukoz/scala-incremental-repro

3. Run `sbt clean compile` in the project
4. Copy the entire directory somewhere else on the disk
5. Enter the newly made directory (all the `target` files should be in there)
6. Run `sbt compile`
7. See the "compiling 1 Scala source to ..." message (which shouldn't be there)

## problem

The code is all being recompiled.

## expectation

Incremental compilation allows using the previously compiled result.

## notes

The same trick with copying works fine without semanticdb enabled. I also found https://github.com/scalameta/scalameta/issues/2053 which might be related (and I guess this would actually be a Zinc issue, but the `semanticdbEnabled` flag is in sbt so I'm reporting here).

Previously, we were using an older version of semanticdb, as well as an older Scala version (2.12.12) which made me think this started happening after the 2.13 upgrade, but it was in fact the newer semanticdb (this same problem happens if you set the `scalaVersion` to 2.12).

We relied on the previous behavior to reuse compilations in Jenkins jobs, by compiling once and copying the whole thing to workspaces used by other jobs (in parallel) - it would be great if we could go back - currently it's hard to work around, as we need semanticdb for semantic Scalafix rules (turning it off and on would _also_ require recompilation).

**Comment by eed3si9n** [comment]
@kubukoz Thanks for the report! I think this is essentially a duplicate of https://github.com/sbt/sbt/issues/6027. See the workaround Yoshida-san posted on the issue as well.

**Comment by kubukoz** [comment]
Thank you @eed3si9n! It looks like the workaround https://github.com/sbt/sbt/issues/6027#issuecomment-717064450 works for me indeed!

**Comment by dabrowski-adam** [comment]
> Thank you [@eed3si9n](https://github.com/eed3si9n)! It looks like the workaround [#6027 (comment)](https://github.com/sbt/sbt/issues/6027#issuecomment-717064450) works for me indeed!

I think that it does not work when using Scala 3 (more specifically, I'm on 3.6.3).
